pkg/sql/colexec/agg/variance/variance.go

```diff
@@ -0,0 +1,221 @@
+			var a float64
+			variance.avgs = append(variance.avgs, a)
+			variance.sizes = append(variance.sizes, 0)
+		}
+	}
+}
+
+func (variance *Variance[T1]) Eval(vs []float64) []float64 {
+	for i, v := range vs {
+		avg := (variance.avgs[i]) / (variance.sizes[i])
+		vs[i] = (v)/(variance.sizes[i]) - math.Pow(float64(avg), 2)
+	}
+	return vs
+}
+
+func (variance *Variance[T1]) Merge(groupIndex1, groupIndex2 int64, x, y float64, IsEmpty1 bool, IsEmpty2 bool, agg any) (float64, bool) {
+	variance2 := agg.(*Variance[T1])
+	if IsEmpty1 && !IsEmpty2 {
+		variance.avgs[groupIndex1] = variance2.avgs[groupIndex2]
+		variance.sizes[groupIndex1] = variance.sizes[groupIndex2]
+		return y, false
+	} else if IsEmpty2 && !IsEmpty1 {
+		return x, false
+	} else if IsEmpty1 && IsEmpty2 {
+		return x, true
+	} else {
+		variance.sizes[groupIndex1] += variance2.sizes[groupIndex2]
+		variance.avgs[groupIndex1] += variance2.avgs[groupIndex2]
+		return (x + y), false
+	}
+}
+
+func (variance *Variance[T1]) Fill(groupIndex int64, v1 T1, v2 float64, z int64, IsEmpty bool, hasNull bool) (float64, bool) {
+	if hasNull {
+		return v2, IsEmpty
+	} else if IsEmpty {
+		f1 := float64(v1)
+		variance.avgs[groupIndex] = any(f1 * float64(z)).(float64)
+		variance.sizes[groupIndex] += float64(z)
+		return math.Pow(f1, 2) * float64(z), false
+	}
+	f1 := float64(v1)
+	f2 := float64(v2)
+	variance.avgs[groupIndex] = float64(variance.avgs[groupIndex]) + f1*float64(z)
+	variance.sizes[groupIndex] += float64(z)
+	return f2 + math.Pow(f1, 2)*float64(z), false
+}
+
+func (variance *Variance2) Grows(sizes int) {
+	if len(variance.avgs) == 0 {
+		variance.avgs = make([]float64, sizes)
+		variance.sizes = make([]float64, sizes)
+	} else {
+		for i := 0; i < sizes; i++ {
+			var a float64
+			variance.avgs = append(variance.avgs, a)
+			variance.sizes = append(variance.sizes, 0)
+		}
+	}
+}
+
+func (variance *Variance2) Eval(vs []float64) []float64 {
+	for i, v := range vs {
+		avg := (variance.avgs[i]) / (variance.sizes[i])
+		vs[i] = (v)/(variance.sizes[i]) - math.Pow(float64(avg), 2)
+	}
+	return vs
+}
+
+func (variance *Variance2) Merge(groupIndex1, groupIndex2 int64, x, y float64, IsEmpty1 bool, IsEmpty2 bool, agg any) (float64, bool) {
+	variance2 := agg.(*Variance2)
+	if IsEmpty1 && !IsEmpty2 {
+		variance.avgs[groupIndex1] = variance2.avgs[groupIndex2]
+		variance.sizes[groupIndex1] = variance.sizes[groupIndex2]
+		return y, false
+	} else if IsEmpty2 && !IsEmpty1 {
+		return x, false
+	} else if IsEmpty1 && IsEmpty2 {
+		return x, true
+	} else {
+		variance.sizes[groupIndex1] += variance2.sizes[groupIndex2]
+		variance.avgs[groupIndex1] += variance2.avgs[groupIndex2]
+		return (x + y), false
+	}
+}
+
+func (variance *Variance2) Fill(groupIndex int64, v1 types.Decimal64, v2 float64, z int64, IsEmpty bool, hasNull bool) (float64, bool) {
+	if hasNull {
+		return v2, IsEmpty
+	} else if IsEmpty {
+		f1, _ := strconv.ParseFloat(String(v1.Decimal64ToString(variance.inputType.Scale)), 64)
```

Comment:
And you are too late, I checked in decimal64/128 stuff and you need to redo this ...     Good thing is that, I think it is easier to use.

Reply:
> And you are too late, I checked in decimal64/128 stuff and you need to redo this ... Good thing is that, I think it is easier to use.

Ok, I'll use your decimal